src/migrations/20250915000000-close-ghost-goal.js

```diff
@@ -0,0 +1,145 @@
+const { prepMigration } = require('../lib/migration');
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.transaction(async (transaction) => {
+      const sessionSig = __filename;
+      await prepMigration(queryInterface, transaction, sessionSig);
+      await queryInterface.sequelize.query(/* sql */`
+        ---------- The steps -------------------------------------------------
+        -- 1: Make sure it's the correct ghost goal
+        -- 2: Complete any In Progress Objectives on that Goal
+        -- 3: actually close the ghost goal
+        ----------------------------------------------------------------------
+
+        -- Just here for validation convenience
+        DROP TABLE IF EXISTS beforestats;
+        CREATE TEMP TABLE beforestats
+        AS
+        SELECT
+          'goals' entity,
+          COUNT(*) FILTER (WHERE status = 'Closed') closed,
+          COUNT(*) FILTER (WHERE status != 'Closed') open
+        FROM "Goals"
+        WHERE "deletedAt" IS NULL
+        UNION
+        SELECT
+          'objectives',
+          COUNT(*) FILTER (WHERE status = 'Complete'),
+          COUNT(*) FILTER (WHERE status != 'Complete')
+        FROM "Objectives"
+        WHERE "deletedAt" IS NULL
+        ORDER BY 1
+        ;
+
+        -- 1: Make sure it's the correct ghost goal
+        DROP TABLE IF EXISTS goals_to_close;
+        CREATE TEMP TABLE goals_to_close
+        AS
+        SELECT DISTINCT
+          g.id gid
+        FROM "Goals" g
+        JOIN "Grants" gr
+          ON g."grantId" = gr.id
+        WHERE g.id = 103372
+          AND gr.number = '90CI010132'
+        ;
+
+        -- 2: Complete any In Progress Objectives on that Goal
+        DROP TABLE IF EXISTS obj_to_complete;
+        CREATE TEMP TABLE obj_to_complete
+        AS
+        SELECT DISTINCT
+          o.id oid,
+          o."otherEntityId" IS NOT NULL other_entity
+        FROM "Objectives" o
+        JOIN goals_to_close
+          ON o."goalId" = gid
+        ;
+        
+        UPDATE "Objectives" o
+        SET
+          status = 'Complete',
+          "updatedAt" = NOW()
+        FROM obj_to_complete
+        WHERE id = oid
+        ;
```

Comment:
Should this restrict to only "In Progress" objectives? 
Right now it will complete all objectives linked to the ghost goal, 
even ones already marked Complete or Not Started.

Reply:
This is a good point. Technically there's only the one Objective on the one goal, but I think it's best to apply the logic correctly so I went ahead and added in the additional predicates